frontend-public/src/components/dashboard/mineInfo/MineInfo.js

```diff
@@ -41,8 +42,30 @@ export class MineInfo extends Component {
   }
 
   handleEditReportSubmit = () => {
+    // Get state of document before latest upload/edits
     const updatedDocument = this.state.selectedDocument;
-    updatedDocument.received_date = moment();
+    // Get document latest version
+    const updatedMineDocument = find(
+      this.props.mine.mine_expected_documents,
+      ({ exp_document_guid }) => exp_document_guid === updatedDocument.exp_document_guid
+    );
+
+    // Set status to received/pending review
+    updatedDocument.exp_document_status.exp_document_status_code = "PRE";
+
+    // Only set received_date when going from 0 to 1 uploaded documents
+    if (
+      updatedDocument.related_documents.length === 0 &&
+      updatedMineDocument.related_documents.length === 1
+    ) {
+      updatedDocument.received_date = moment();
+    }
```

Comment:
This fails to set the date if you upload two files and hit submit.

Reply:
Ah... I missed that use-case. I'll rethink this solution

Reply:
Updated the logic to handle multiple documents uploaded before a submission